programs/wen-new-standard/src/instructions/mint/add_metadata.rs

```diff
@@ -0,0 +1,70 @@
+use std::str::FromStr;
+
+use anchor_lang::{prelude::*, solana_program::entrypoint::ProgramResult};
+
+use anchor_spl::token_interface::{
+    spl_token_metadata_interface::state::Field, token_metadata_update_field, Mint, Token2022,
+    TokenMetadataUpdateField, TokenMetadataUpdateFieldArgs,
+};
+
+use crate::{errors::MetadataErrors, update_account_lamports_to_minimum_balance2};
+
+#[derive(AnchorDeserialize, AnchorSerialize)]
+pub struct AddMetadataArgs {
+    pub field: String,
+    pub value: String,
+}
+
+#[derive(Accounts)]
+pub struct AddMetadata<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    #[account(
+        mut,
+        mint::token_program = token_program,
+    )]
+    pub mint: Box<InterfaceAccount<'info, Mint>>,
+    pub system_program: Program<'info, System>,
+    pub token_program: Program<'info, Token2022>,
+}
+
+impl<'info> AddMetadata<'info> {
+    fn update_token_metadata_field(&self, field: Field, value: String) -> ProgramResult {
+        let cpi_accounts = TokenMetadataUpdateField {
+            token_program_id: self.token_program.to_account_info(),
+            metadata: self.mint.to_account_info(),
+            update_authority: self.authority.to_account_info(),
+        };
+        let cpi_ctx = CpiContext::new(self.token_program.to_account_info(), cpi_accounts);
+        token_metadata_update_field(cpi_ctx, TokenMetadataUpdateFieldArgs { field, value })?;
```

Comment:
is there any validation to be done on the length of this account? not sure what happens if this dictionary fills up or if this instruction automatically lengthens it.

Reply:
I think the instruction automatically lengthens it and that's why we need to allocate more space